modules/generator/instance.go

```diff
@@ -111,6 +112,16 @@ func newInstance(cfg *Config, instanceID string, overrides metricsGeneratorOverr
 		logger: logger,
 	}
 
+	// NOTE: When a new histogram is created, the overrides are checked to determine
+	// which histogram implementation to use.  This happens when the servicegraphs
+	// or spanmetrics processors are created.  The new implementation can
+	// generate native histograms or both classic and native.  However, if the
+	// new implementation is chosen, and then the overrides are updated to
+	// indicated that the old implementation should be use by setting "classic",
+	// we need to know if the processors should be re-created.  Store the current
+	// value for later comparison.
+	i.histogramsOverride.Store(i.overrides.MetricsGeneratorGenerateNativeHistograms(i.instanceID))
```

Comment:
Do we need this `Store()`? Since we call `updateProcessors` immediately after this we store it there anyways.

We can maybe drop this line and move the NOTE to `updateProcessors`?

Reply:
Yep, your right.  Since the initial processors will be empty, we store the value before we need it on the next iteration.